### DISCUSSION TITLE
FreeU Unsupported dtype BFloat16.

### DISCUSSION THREAD

**Discussion by pshtif** [huggingface/diffusers#6773] [Q&A] [answered]
I am curious is this a limitation of pytorch or a problem within diffusers that once you switch models to BFloat16 it is no longer possible to combine it with FreeU.

**Comment by sayakpaul** [comment]
Do you have a reproducible code snippet?

**Comment by pshtif** [comment]
Basically just using FreeU as proposed here https://huggingface.co/docs/diffusers/using-diffusers/freeu but switching torch dtype to bfloat16

**Reply by sayakpaul** [reply, marked as answer]
Could be because internally FreeU uses Fourier transforms and it might not have sound behaviour in BFloat16.

**Reply by pshtif** [reply]
That is what I thought, that it is not really a model nor diffusers issue. Maybe a better handling of such error would be handy. Anyway thanks for giving out the missing info.
